service/lint.ml

```diff
@@ -0,0 +1,19 @@
+open Current.Syntax
+module Docker = Current_docker.Default
+
+let format_dockerfile ~base ~ocamlformat_version =
+  let open Dockerfile in
+  from (Docker.Image.hash base)
+  @@ run "git -C /home/opam/opam-repository pull"
+  @@ run "opam depext ocamlformat=%s" ocamlformat_version
+  @@ run "opam install ocamlformat=%s" ocamlformat_version
+
+let v_from_opam ~ocamlformat_version ~base ~src =
+  let dockerfile =
+    let+ base = base and+ ocamlformat_version = ocamlformat_version in
+    format_dockerfile ~base ~ocamlformat_version
+  in
+  let img =
+    Docker.build ~label:"OCamlformat" ~pull:false ~dockerfile (`Git src)
+  in
+  Docker.run ~label:"lint" img ~args:[ "dune"; "build"; "@fmt" ]
```

Comment:
This doesn't appear to get run with access to the source code, which seems strange. Maybe it should just be another build step?

Reply:
Not sure what you mean here; the source is built into the image on the previous step, no?

Reply:
No, it's just made available in the context. But there are no `COPY` steps, so it isn't used for anything. Do you have an example of the check failing on something?

Reply:
Of course you're right. I managed to refactor away a `copy` step because I didn't understand what it was doing... have now included it again.